File: frontend/src/schema.ts
export type DirItemBase = {
  size: number;
  last_modified: string;
  is_dir: boolean;
};

export interface FolderDirItem extends DirItemBase {
  is_dir: true;
  folder_path: string;
  folder_name: string;
}

export interface FileDirItem extends DirItemBase {
  is_dir: false;
  file_name: string;
  file_path: string;
}

export type DirItemFromServer = FolderDirItem | FileDirItem;
export type ListDirResult = {
  dirent_list?: DirItemFromServer[];
};

export interface DirItemClient extends DirItemBase {
  name: string;
  path: string;
}

File: frontend/src-tauri/src/schema.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct DirItem {
    size: u128,
    is_dir: bool,
    last_modified: String,
    pub(crate) file_path: Option<String>,
    file_name: Option<String>,
    folder_name: Option<String>,
    pub(crate) folder_path: Option<String>,
}

pub type DirList = Vec<DirItem>;

#[derive(Serialize, Deserialize)]
pub struct ListData {
    pub(crate) dirent_list: Option<DirList>,
}
